### ISSUE TITLE
Android Sync

### ISSUE THREAD

**Issue by AccursedAngel** [Alex313031/Thorium-Android#6]
**System Details**
 - OS: Android 13 Stable (Sept. Security Update) (Google Pixel 6a)
 - Thorium Version: 116.0.5845.184

**Problem**
After downloading the ARM 64 version and installing it, it prompts me to add a new account instead of the one currently present on the device. I obviously cannot add an account already on the device.
**Screenshots**
![image](https://github.com/Alex313031/Thorium-Android/assets/45458291/806e9aad-89dc-47ba-9893-7e06ffd09be1)

**Additional Notes**

**Comment by Alex313031** [comment]
@AccursedAngel This is a known issue with the Android version. Unlike the Desktop versions, where just adding API Keys is sufficient to enable Sync, I believe with Chromium on Android there is an additional proprietary component that I don't have access to that is needed. Case in point, there are very few Chromium based browsers for Android with Sync supported. I don't know how to fix this, but I can ask in the chromium-dev google group for help.

**Comment by AccursedAngel** [comment]
Ah I gotcha, thanks for replying. Sorry about the duplicate posts here and on Reddit. <3

**Comment by aniruddhgutta** [comment]
hey, did you receive any updates?

**Comment by gz83** [comment]
[bugs.chromium.org/p/chromium/issues/detail?id=677887](https://bugs.chromium.org/p/chromium/issues/detail?id=677887)

Google restricts login and synchronization of third-party versions, and there is currently no solution

@AccursedAngel @aniruddhgutta

**Comment by DragonHeart69** [comment]
Hey,
Do you see any possibility of having your own sync server, or being able to use a self-hosted sync server ?
Maybe a startingpount: https://community.brave.com/t/self-hosted-sync-server/305806

**Comment by gz83** [comment]
There should be no plans to build a synchronization server at the moment.

@DragonHeart69

**Comment by ghost** [comment]
why is this closed? this is ongoing

**Comment by gz83** [comment]
> why is this closed? this is ongoing

What was closed?

@komprimiert

**Comment by ghost** [comment]
Can't you guys implement something like brave? brave doesn't use google sync ry8

**Comment by gz83** [comment]
Is there any solution to implement a custom synchronization server at this stage?  If there are relevant solutions on the Internet, please provide them to me here and I will consider porting the relevant functions to Thorium.

If the Thorium team is required to deploy a synchronization server, we may not currently have sufficient funds to do this work.

Thorium is different from Brave. The Brave browser is supported by a company and a large amount of advertising revenue, and it also has more developers than Thorium. Please understand the different situations of the two.

@Subhashis2007

**Comment by ghost** [comment]
The only "solution" I have seen which I haven't tried yet is xbrowsersync which was suggested on reddit

**Comment by ghost** [comment]
> Is there any solution to implement a custom synchronization server at this stage? If there are relevant solutions on the Internet, please provide them to me here and I will consider porting the relevant functions to Thorium.
> 
> If the Thorium team is required to deploy a synchronization server, we may not currently have sufficient funds to do this work.
> 
> Thorium is different from Brave. The Brave browser is supported by a company and a large amount of advertising revenue, and it also has more developers than Thorium. Please understand the different situations of the two.
> 
> @Subhashis2007

got it,  well I hope so more and more developers join this, and may be giving an option whether to enable sponsored images on homescreen to generate some funds would be great too, in this way the users can contribute

**Comment by DragonHeart69** [comment]
@gz83 for me, a selfhosted docker is good. The link to the brave selfhosted is in my other comment. I understand that you can't host a server for all the users.

**Comment by gz83** [comment]
OK, I'll take a look

@DragonHeart69

**Comment by ghost** [comment]
> @gz83 for me, a selfhosted docker is good. The link to the brave selfhosted is in my other comment. I understand that you can't host a server for all the users.

can you link the comment? I'm interested as well

**Comment by gz83** [comment]
> > @gz83 for me, a selfhosted docker is good. The link to the brave selfhosted is in my other comment. I understand that you can't host a server for all the users.
> 
> can you link the comment? I'm interested as well

https://community.brave.com/t/self-hosted-sync-server/305806

**Comment by DumbledoreCalrissian** [comment]
Would some kind of implementation of MicroG help?
(like how ReVanced YouTube app lets you sign into google)

**Comment by GalaxyXYZ888** [comment]
Yeah this is a very big problem in the Android version. Sync in all devices is very important for me at least. But its normal this version is still without extensions and other features also. The MacOS and Windows versions are very very good and I will probably live without the sync at the time being XD

**Comment by Spinnekop** [comment]
Any way to import a bookmarks file?

**Comment by ibay770** [comment]
> @AccursedAngel This is a known issue with the Android version. Unlike the Desktop versions, where just adding API Keys is sufficient to enable Sync, I believe with Chromium on Android there is an additional proprietary component that I don't have access to that is needed. Case in point, there are very few Chromium based browsers for Android with Sync supported. I don't know how to fix this, but I can ask in the chromium-dev google group for help.

Thats fair. Is it possible to add support for extensions instead? Maybe someone will make an extension to another cloud storage provider like OneDrive and we can add it to Thorium

**Comment by PythonesqueSpam** [comment]
Rather than using Google sync or creating a Thorium sync server with all the overhead that entails, you could allow the use of the customer's own online storage, perhaps through OneDrive, Box, Google Drive , Dropbox, etc. with Thorium data being stored there, perhaps in an encrypted form and used to synchronise between platforms?  My password manager app uses this sort of mechanism to sync between my Phone and Desktop computer, so avoiding the need to make a sync server.

**Comment by FadeMind** [comment]
Hi Devs 
First launch Thorium Mobile prompt login to Google account with Passkey fingerprint. Validate do nothing. Adding Google Account via e-mail make error - e-mail address is already in use on this device. 

Also installing your version of Android Web View and disabling built in system version of Android Web View make Play store components crash. I can create separate issue if needed. Regards

**Comment by avichou** [comment]
can we except anything on this or is it dead in the water ?

**Comment by chengnan049** [comment]
7 months have passed and this problem has not been solved

**Comment by ShreyasJejurkar** [comment]
Still not solved. :(

**Comment by TheBense** [comment]
Would it be possible to have Thorium sync with some a service _other_ than google chrome?

Perhaps something like being able to sync to firefox ? Not sure how feasible this would be.

**Comment by polymo1** [comment]
> Would it be possible to have Thorium sync with some a service _other_ than google chrome?
> 
> Perhaps something like being able to sync to firefox ? Not sure how feasible this would be.

It could possibly entail a complete rewrite of the existing sync system.

**Comment by TheBense** [comment]
So?

**Comment by polymo1** [comment]
> So?

I'd imagine that's a lot of extra work with little benefit when compared to switching to something like MicroG.

**Comment by ybgmoli** [comment]
Youtube revanced also has this problem on unrooted devices. wouldnt it be possible to create a module for magisk so us, rooted users can log in into thorium?

**Comment by polymo1** [comment]
> Youtube revanced also has this problem on unrooted devices. wouldnt it be possible to create a module for magisk so us, rooted users can log in into thorium?

YouTube revanced + MicroG allows signin

**Comment by ybgmoli** [comment]
> > Youtube revanced also has this problem on unrooted devices. wouldnt it be possible to create a module for magisk so us, rooted users can log in into thorium?
> 
> YouTube revanced + MicroG allows signin

yh I know but the dev said that he won't be implementing microg any time soon.

**Comment by RubenKelevra** [comment]
How about creating a pledge goal for this?

**Comment by GalaxyXYZ888** [comment]
> How about creating a pledge goal for this?

Shut up and take my money approach.... I like it ! 😁

**Comment by schmekism** [comment]
> Rather than using Google sync or creating a Thorium sync server with all the overhead that entails, you could allow the use of the customer's own online storage, perhaps through OneDrive, Box, Google Drive , Dropbox, etc. with Thorium data being stored there, perhaps in an encrypted form and used to synchronise between platforms? My password manager app uses this sort of mechanism to sync between my Phone and Desktop computer, so avoiding the need to make a sync server.

@Alex313031 is this a reasonable interim solution? 

@PythonesqueSpam  what password manager are you using to access synced bookmarks that uses this "mechanism?"

**Comment by gz83** [comment]
Third-party Chromium fork browsers are not allowed to use Google's native account login process. In the new version, I have managed to hide the relevant entry point.
